bgpd/bgp_io.c

```diff
@@ -200,8 +203,18 @@ static int read_ibuf_work(struct peer_connection *connection)
 	stream_set_endp(pkt, pktsize);
 
 	frrtrace(2, frr_bgp, packet_read, connection->peer, pkt);
+
+	/* get the type of message */
+	stream_forward_getp(pkt, BGP_MARKER_SIZE + 2);
```

Comment:
Please write some code that ensures that the pktsize is *sufficient* to actually read this data.